### PR TITLE
Add ProprietaryBankTransactionCode <BkTxCd>.<Prtry> to the entry

### PR DESCRIPTION
**Pull request by bdudelsack**

### CODE REVIEW

**Comment by Einenlum** [comment]
@bdudelsack Need rebase.

**Comment by bdudelsack** [comment]
rebased

**Comment by Einenlum** [comment]
It needs rebase again ^^

**Comment by bdudelsack** [comment]
done

File: src/DTO/BankTransactionCode.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+/**
```
Comment (Einenlum) [inline src/DTO/BankTransactionCode.php]: This is generated by your tool. These comments should not end in the project.

File: src/DTO/BankTransactionCode.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Created by IntelliJ IDEA.
+ * User: bdudelsack
+ * Date: 29.08.16
+ * Time: 13:17
+ */
+
+namespace Genkgo\Camt\DTO;
+
```
Comment (Einenlum) [inline src/DTO/BankTransactionCode.php]: extraline

File: src/DTO/BankTransactionCode.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
```
Comment (Einenlum) [inline src/DTO/BankTransactionCode.php]: extraline

File: src/DTO/BankTransactionCode.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    public function getProprietary()
+    {
+        return $this->proprietary;
+    }
+
+    /**
+     * @param ProprietaryBankTransactionCode $proprietary
+     */
+    public function setProprietary($proprietary)
```
Comment (Einenlum) [inline src/DTO/BankTransactionCode.php]: No typehint?

File: src/DTO/BankTransactionCode.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        $this->proprietary = $proprietary;
+    }
+}
```
Comment (Einenlum) [inline src/DTO/BankTransactionCode.php]: Missing extra blank line

File: src/DTO/Entry.php [hunk 6]
```diff
@@ -56,6 +56,10 @@ class Entry
      * @var string
      */
     private $additionalInfo;
+    /**
```
Comment (Einenlum) [inline src/DTO/Entry.php]: missing extra blank line

File: src/DTO/Entry.php [hunk 7]
```diff
@@ -222,4 +226,20 @@ public function setAdditionalInfo($additionalInfo)
...
+     */
+    public function getBankTransactionCode()
+    {
+        return $this->bankTransactionCode;
+    }
+
+    /**
+     * @param BankTransactionCode $bankTransactionCode
+     */
+    public function setBankTransactionCode($bankTransactionCode)
```
Comment (Einenlum) [inline src/DTO/Entry.php]: No typehint?

File: src/DTO/EntryTransactionDetail.php [hunk 8]
```diff
@@ -175,4 +180,20 @@ public function getAdditionalTransactionInformation()
...
+     */
+    public function getBankTransactionCode()
+    {
+        return $this->bankTransactionCode;
+    }
+
+    /**
+     * @param BankTransactionCode $bankTransactionCode
+     */
+    public function setBankTransactionCode($bankTransactionCode)
```
Comment (Einenlum) [inline src/DTO/EntryTransactionDetail.php]: missing typehint

File: src/DTO/ProprietaryBankTransactionCode.php [hunk 9]
```diff
@@ -0,0 +1,61 @@
+<?php
+/**
```
Comment (Einenlum) [inline src/DTO/ProprietaryBankTransactionCode.php]: Same

File: src/DTO/ProprietaryBankTransactionCode.php [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+ * Created by IntelliJ IDEA.
+ * User: bdudelsack
+ * Date: 29.08.16
+ * Time: 13:18
+ */
+
+namespace Genkgo\Camt\DTO;
+
+
```
Comment (Einenlum) [inline src/DTO/ProprietaryBankTransactionCode.php]: extraline

File: src/DTO/ProprietaryBankTransactionCode.php [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+class ProprietaryBankTransactionCode
+{
+    /** @var string */
+    private $code;
+    /** @var string */
+    private $issuer;
+
+    /**
+     * ProprietaryBankTransactionCode constructor.
```
Comment (Einenlum) [inline src/DTO/ProprietaryBankTransactionCode.php]: useless comment

File: src/DTO/ProprietaryBankTransactionCode.php [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    /**
+     * @param string $issuer
+     */
+    public function setIssuer($issuer)
+    {
+        $this->issuer = $issuer;
+    }
+}
```
Comment (Einenlum) [inline src/DTO/ProprietaryBankTransactionCode.php]: missing extra blank line

File: tests/Unit/Camt052/EndToEndTest.php [hunk 13]
```diff
@@ -229,4 +229,39 @@ public function testRelatedParties()
...
+
+                    $this->assertEquals('XXXX+000+0000+000', $entry->getBankTransactionCode()->getProprietary()->getCode());
+                    $this->assertEquals('ZKA', $entry->getBankTransactionCode()->getProprietary()->getIssuer());
+
+                    $this->assertCount(1, $entry->getTransactionDetails());
+                    foreach($entry->getTransactionDetails() as $details) {
+                        $this->assertInstanceOf(DTO\BankTransactionCode::class, $details->getBankTransactionCode());
+                        $this->assertInstanceOf(DTO\ProprietaryBankTransactionCode::class, $details->getBankTransactionCode()->getProprietary());
+
+                    $this->assertEquals('XXXX+000+0000+000', $details->getBankTransactionCode()->getProprietary()->getCode());
```
Comment (Einenlum) [inline tests/Unit/Camt052/EndToEndTest.php]: bad indent

**Comment by Einenlum** [comment]
After fixing these comments you can also use `$ ./vendor/bin/php-cs-fixer fix` to fix potential CS issues :).

**Comment by bdudelsack** [comment]
Should there be a blank line on the end of each PHP file?

**Comment by Einenlum** [comment]
It's not about PHP, but a general POSIX rule (that's why Github always adds a red arrow at the end of your files which do not respect it). You can check these links to understand more about it : 

http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline
http://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file
https://robots.thoughtbot.com/no-newline-at-end-of-file

It seems you're using Intellij? You can check here how to automate this: http://stackoverflow.com/questions/16761227/how-to-make-intellij-idea-insert-a-new-line-at-every-end-of-file

**Comment by Einenlum** [comment]
And this typehint is still missing :) https://github.com/genkgo/camt/pull/31#discussion_r76944475

**Comment by bdudelsack** [comment]
What about squashing the commits? Should i squash them before the branch get merged?

**Comment by Einenlum** [comment]
@bdudelsack I squashed them directly on Github :). Thanks!

**Pull request merged**